marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs

```diff
@@ -592,7 +592,7 @@ class FlowUtils {
         }
       }
       docHeaders = docHeadersArray.reduce((acc, cur) => Object.assign(acc,cur), {});
-      headers = Object.assign({}, headers, docHeaders);
+      headers = Object.assign({}, docHeaders, headers);
```

Comment:
This definitely works when creating a new document. If a user is overwriting an existing document (perhaps the source document itself) - this still seems correct because we're really creating a brand new document from the perspective of the mapping step - i.e. we're overwriting a source document with an entity instance, and those are two totally different things. So overwriting createdOn/createdBy makes sense. 

Couple thoughts to make this easier to understand:

- Rename "headers" to "headersFromStepOptions" to make it more clear what they're describing (they could be interpreted as headers from a source document)
- I think we should swap the order of the first two args in the function so that it is consistent with the order in the "assign" call. And the function needs comments on it to make it clear that the output will be the result of merging "headersFromStepOptions" into "docHeaders", where the former will overwrite the latter where there are duplicate keys. 

Finally, we need a unit test for this function. The HC test is fine, but that's a very high level test for a low level detail like this. The unit test won't care about steps - it'll just pass in in-memory args. Should include a scenario for XML as well so we have test coverage there.